tests/cli_ui_test.py

```diff
@@ -36,136 +36,189 @@
-                },
-            ],
-            {"prompt_ret": {"wants_to_document_model": True, "model_description": ""}},
-        ),
-        (
-            "undocumented_columns",
+    ],
+)
+def test__document_model(mocker, question_payload, questionary_outputs, expected_results):
+    from dbt_sugar.core.ui.cli_ui import UserInputCollector
+
+    print(question_payload)
+    # mock for the questionary.Question --this was a bit of a fuckery to test...
+    # class Question:
+    #     def __init__(self, return_value):
+    #         self._return_value = return_value
+
+    #     def ask(self):
+    #         return self._return_value
+
+    mocker.patch("questionary.prompt", return_value=questionary_outputs)
+    results = UserInputCollector(question_type="model", question_payload=question_payload).collect()
+    if questionary_outputs.get("wants_to_document_model") is False:
+        assert results == dict()
+    else:
+        assert results == expected_results
+
+
+@pytest.mark.parametrize(
+    "question_payload, questionary_outputs, expected_results",
+    [
+        pytest.param(
             [
                 {
                     "type": "checkbox",
                     "name": "cols_to_document",
                     "choices": ["col_a", "col_b"],
                     "message": "Select the columns you want to document.",
-                }
+                },
             ],
             {
-                "confirm_ret": True,
-                "prompt_ret": {"col_a": "Custom desc", "col_b": "Custom desc"},
-                "text_ret": "Custom desc",
+                "confirm_return": True,
+                "prompt_return": {"col_a": "Custom desc", "col_b": "Custom desc"},
             },
+            {"col_a": {"description": "Custom desc"}, "col_b": {"description": "Custom desc"}},
+            id="document_all_undocumented",
         ),
-        (
-            "undocumented_columns",
+        pytest.param(
             [
                 {
                     "type": "checkbox",
                     "name": "cols_to_document",
                     "choices": ["col_a", "col_b"],
                     "message": "Select the columns you want to document.",
-                }
+                },
             ],
             {
-                "confirm_ret": False,
-                "prompt_ret": {"col_a": "Custom desc"},
-                "non_full_cols": {"cols_to_document": ["col_a"]},
-                "text_ret": "Custom desc",
+                "confirm_return": False,
+                "prompt_return": {"cols_to_document": ["col_a"]},
             },
+            {"col_a": {"description": "Custom desc"}},
+            id="document_only_some_columns",
         ),
-        (
-            "documented_columns",
+    ],
+)
+def test__document_undocumented_columns(
+    mocker, question_payload, questionary_outputs, expected_results
+):
+    from dbt_sugar.core.ui.cli_ui import UserInputCollector
```

Comment:
if you are importing this in all the methods, will not be better to put it in the top and avoid duplication, you can do the same with the class Question

Reply:
Yeah that makes sense indeed. Will change now.